### PR TITLE
[INJIMOB-1861] add support for mso_mdoc VC

### PR DESCRIPTION
**Pull request by KiruthikaJeyashankar**
## Description

> Ability to download, view and share with verifier over BLE (w/o vc verification happening in verifier side) mso_mdoc vc

Other major changes:
- Identification of the VC's respective credential issuer metadata from well known exposed is happening via the credential configuration id (which is the key of the element in credentials_configurations_supported object in the wellknown)

## Issue ticket number and link

> [INJIMOB-1861](https://mosip.atlassian.net/browse/INJIMOB-1861&#41)

## Screenshots

**_Mini view_**
image

**_History_**
image

**_Detail view_**
image

[INJIMOB-1861]: https://mosip.atlassian.net/browse/INJIMOB-1861?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: android/app/src/main/java/io/mosip/residentapp/InjiVciClientModule.java [hunk 1]
```diff
@@ -42,19 +42,33 @@ public String getName() {
     public void requestCredential(ReadableMap issuerMetaData, String jwtProofValue, String accessToken, Promise promise) {
         try {
             CredentialFormat credentialFormat;
+            IssuerMetaData constructedIssuerMetadata ;
             switch (issuerMetaData.getString("credentialFormat")) {
                 case "ldp_vc":
```
Comment (sree96) [inline android/app/src/main/java/io/mosip/residentapp/InjiVciClientModule.java:47]: Lets move these to constants

File: android/app/src/main/java/io/mosip/residentapp/InjiVciClientModule.java [hunk 2]
```diff
@@ -42,19 +42,33 @@ public String getName() {
...
-            CredentialResponse response = vciClient.requestCredential(new IssuerMetaData(
+                    constructedIssuerMetadata =  new IssuerMetaData(
                             issuerMetaData.getString("credentialAudience"),
                             issuerMetaData.getString("credentialEndpoint"),
                             issuerMetaData.getInt("downloadTimeoutInMilliSeconds"),
                             convertReadableArrayToStringArray(issuerMetaData.getArray("credentialType")),
-                            credentialFormat), new JWTProof(jwtProofValue)
+                            credentialFormat,null,null);
+                    break;
+                case "mso_mdoc":
```
Comment (sree96) [inline android/app/src/main/java/io/mosip/residentapp/InjiVciClientModule.java:56]: Lets move these to constants

**Review by sree96 — CHANGES_REQUESTED** [review]
Address the couple of comments added

**Review by abhip2565 — CHANGES_REQUESTED** [review]

File: android/app/src/main/java/io/mosip/residentapp/InjiVciClientModule.java (hunk 2 above)
Comment (KiruthikaJeyashankar) [reply android/app/src/main/java/io/mosip/residentapp/InjiVciClientModule.java]: Modified to iff else for removing harcoding as switch case is having issues using Credential format enum constant values

**Review by sree96 — APPROVED** [review]

**Pull request merged**